.github/workflows/scripts/e2e.generic.default.verify.sh

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env bash
+
+source "./.github/workflows/scripts/e2e-utils.sh"
+
+go env -w GOFLAGS=-mod=mod
+
+# Install from HEAD
+go install github.com/slsa-framework/slsa-verifier@latest
+
+BRANCH=$(echo "$THIS_FILE" | cut -d '.' -f4)
+
+echo "branch is $BRANCH"
+echo "GITHUB_REF_NAME: $GITHUB_REF_NAME"
+echo "GITHUB_REF_TYPE: $GITHUB_REF_TYPE"
+echo "GITHUB_REF: $GITHUB_REF"
+
+# Default parameters.
+if [[ "$BRANCH" == "main" ]]; then
+    slsa-verifier --artifact-path "$BINARY" --provenance "$PROVENANCE" --source "github.com/$GITHUB_REPOSITORY"
+    e2e_assert_eq "$?" "0" "main default parameters"
+else
+    slsa-verifier --artifact-path "$BINARY" --provenance "$PROVENANCE" --source "github.com/$GITHUB_REPOSITORY"
+    e2e_assert_not_eq "$?" "0" "not main default parameters"
+fi
+
+echo "DEBUG: file is $THIS_FILE"
+
+# Correct branch
+slsa-verifier --branch "$BRANCH" --artifact-path "$BINARY" --provenance "$PROVENANCE" --source "github.com/$GITHUB_REPOSITORY"
+e2e_assert_eq "$?" "0" "should be branch $BRANCH"
+
+# Wrong branch
+slsa-verifier --branch "not-$GITHUB_REF_NAME" --artifact-path "$BINARY" --provenance "$PROVENANCE" --source "github.com/$GITHUB_REPOSITORY"
+e2e_assert_not_eq "$?" "0" "wrong branch"
+
+# Wrong tag
+slsa-verifier --tag v1.2.3 --artifact-path "$BINARY" --provenance "$PROVENANCE" --source "github.com/$GITHUB_REPOSITORY"
+e2e_assert_not_eq "$?" "0" "wrong tag"
+
+if [[ "$GITHUB_REF_TYPE" == "tag" ]]; then
```

Comment:
in a follow-up PR, let's create function for these, since they are the same as the SLSA3 Go. This way we can have functions that apply to all builders.